From 224a1d239d6a4f30513ee66a36b91d14f9363717 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 26 Jan 2020 00:40:13 +0100
Subject: [PATCH] media: sun6i-csi: Fix vidioc_g/s_parm to handle lack of the
 sensor

In case no link is enabled, sun6i_video_remote_subdev would return NULL.
Handle this gracefully.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c b/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c
index 5b7cdef3bcc3e..b05fbe22c430e 100644
--- a/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c
+++ b/drivers/media/platform/sunxi/sun6i-csi/sun6i_video.c
@@ -447,6 +447,8 @@ static int vidioc_g_parm(struct file *file, void *priv,
 	struct v4l2_subdev *subdev;
 
 	subdev = sun6i_video_remote_subdev(video, NULL);
+	if (!subdev)
+		return -ENXIO;
 
 	return v4l2_g_parm_cap(video_devdata(file), subdev, p);
 }
@@ -458,6 +460,8 @@ static int vidioc_s_parm(struct file *file, void *priv,
 	struct v4l2_subdev *subdev;
 
 	subdev = sun6i_video_remote_subdev(video, NULL);
+	if (!subdev)
+		return -ENXIO;
 
 	return v4l2_s_parm_cap(video_devdata(file), subdev, p);
 }
-- 
2.37.2

